runtime/queries/column_timeseries.go

```diff
@@ -432,6 +444,10 @@ func getFilterFromDimensionValuesFilter(
 					likeClause += " OR "
 				}
 			}
+			if !nulls && exclude {
+				// In case of exclusion, we need to explicitly include NULL value
+				likeClause = fmt.Sprintf("%s OR %s IS NULL", likeClause, escapedName)
+			}
 			conditions = append(conditions, likeClause)
 		}
 		result += strings.Join(conditions, " "+dimensionJoiner+" ")
```

Comment:
1. As mentioned for metrics views, add the `OR %s` IS NULL` once instead of in every condition. 
2. `dimensionJoiner` can be an `"AND"`, which binds tighter than the `OR`s in the `likeClause`